Lib/defconQt/glyphView.py

```diff
@@ -1380,68 +1682,127 @@ def __init__(self, glyph, parent=None):
+        self.addAnchors()
+        self.addPoints()
 
-    # TODO: diagnose notifications count
-    def _layersChanged(self, notification):
-        self.redrawGlyph()
-        self.redrawOtherLayers()
+    def updateLayerAssets(self):
+        for item in self._getSceneItems('hMetricLabels'):
+            item.setPos(self._glyph.width, item.y())
 
-    def redrawGlyph(self):
-        path = self._glyph.getRepresentation("defconQt.NoComponentsQPainterPath")
+    def activeGlyphChanged(self):
+        # For now, we'll assume not scene._blocked == moving UI points
+        # this will not be the case anymore when drag sidebearings pops up
         scene = self.scene()
-        scene._outlineItem.setPath(path)
-        if not scene._blocked:
-            # TODO: also rewind anchors and components
-            for item in scene.items():
-                if isinstance(item, (OnCurvePointItem, ComponentItem, AnchorItem)):
-                    scene.removeItem(item)
-                elif isinstance(item, VGuidelinesTextItem):
-                    item.setPos(self._glyph.width, item.y())
+        if scene._blocked:
+            self.updateActiveLayerPath()
+            return
+        self.updateActiveLayer();
 
-            self.addComponents()
-            self.addAnchors()
-            self.addPoints()
-            # For now, we'll assume not scene._blocked == moving UI points
-            # this will not be the case anymore when drag sidebearings pops up
-            scene._widthItem.setRect(0, -1000, self._glyph.width, 3000)
+    def updateActiveLayer(self):
+        self._getSceneItems('scene-added-items', clear=True)
+        self.updateLayerAssets()
+
+        self.updateActiveLayerPath()
+        self.addComponents()
+        self.addAnchors()
+        self.addPoints()
 
-    def redrawOtherLayers(self):
+        # this is related to addBackground
         scene = self.scene()
-        for item in scene.items():
-            # XXX: discriminate better
-            if isinstance(item, QGraphicsPathItem) and item.zValue() == -997:
-                scene.removeItem(item)
-        self.addOtherLayersOutlines()
+        scene._widthItem.setRect(0, -1000, self._glyph.width, 3000)
 
     def addBackground(self):
         scene = self.scene()
         font = self._glyph.getParent()
         width = self._glyph.width
-        if width is None: width = 0
         item = scene.addRect(-1000, -1000, 3000, 3000, QPen(Qt.black), QBrush(Qt.gray))
         item.setZValue(-1000)
         scene._widthItem = scene.addRect(0, -1000, width, 3000, QPen(Qt.NoPen), QBrush(backgroundColor))
         scene._widthItem.setZValue(-999)
-        descender = font.info.descender
-        if descender is None: descender = -250
-        unitsPerEm = font.info.unitsPerEm
-        if unitsPerEm is None: unitsPerEm = 1000
-        self.centerOn(width/2, descender+unitsPerEm/2)
+        self.centerOn(width/2, font.info.descender+font.info.unitsPerEm/2)
```

Comment:
right, when I did this it was pre:

https://github.com/trufont/trufont/commit/a80988874c5d621a2b11c75f5af31b23d3008ee7#diff-4afada8b45ba97a28f2ff0c5a9995dc0L1420

Sorry for this. I did some hand merging and missed this :-/